File: 240311/행복한 수열의 개수/number-of-happy-sequence.cpp
#include <bits/stdc++.h>
using namespace std;

int n, m, arr[101][101], ans;

void input() {
    cin >> n >> m;
    for (int i=1; i<=n; i++) {
        for (int j=1; j<=n; j++) {
            cin >> arr[i][j];
        }
    }
}

int checkRow(int y) {
    int cnt = 0;

    int tmp = 1;
    for (int i=2; i<=n; i++) {
        if (arr[y][i] == arr[y][i-1]) tmp++;
        else {
            cnt = max(cnt, tmp);
            tmp = 1;
        }
    }
    cnt = max(cnt, tmp);

    if (cnt >= m) return 1;
    else return 0;
}

int checkColumn(int x) {
    int cnt = 0;

    int tmp = 1;
    for (int i=2; i<=n; i++) {
        if (arr[i][x] == arr[i-1][x]) tmp++;
        else {
            cnt = max(cnt, tmp);
            tmp = 1;
        }
    }
    cnt = max(cnt, tmp);

    if (cnt >= m) return 1;
    else return 0;
}

void solve() {
    for (int i=1; i<= n; i++) {
        ans += checkRow(i);
        ans += checkColumn(i);
    }
    cout << ans;
}

int main() {
    input();
    solve();
    return 0;
}

File: 240408/행복한 수열의 개수/number-of-happy-sequence.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const [n, m] = input[0].split(' ').map(Number);
const arr = input.slice(1, 1 + n).map(line => line.trim().split(' ').map(Number));

let ans = 0;

// 각 행마다 연속인 숫자의 개수 세기
for (let y = 0; y < n; y++) {
    let cnt = 1; let tmp = 1;
    for (let x = 1; x < n; x++) {
        if (arr[y][x] === arr[y][x - 1]) tmp++;
        else tmp = 1;
        cnt = Math.max(cnt, tmp);
    }
    if (cnt >= m) ans++;
}

for (let x = 0; x < n; x++) {
    let cnt = 1; let tmp = 1;
    for (let y = 1; y < n; y++) {
        if (arr[y][x] === arr[y - 1][x]) tmp++;
        else tmp = 1;
        cnt = Math.max(cnt, tmp);
    }
    if (cnt >= m) ans++;
}

console.log(ans);